### PR TITLE
Remove the deprecated load_ocean_ridge_points function (deprecated since v0.6.0) 

### PR DESCRIPTION
**Pull request by willschlitzer**
This replaces `load_ocean_ridge_points` with `_load_ocean_ridge_points`.

It also changes the argument `sep=r"\s+"` to `delim_whitespace=True`.

Addresses: #2302

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -142,39 +142,25 @@ def load_japan_quakes(**kwargs):
...
     """
-    (Deprecated) Load a table of ocean ridge points for the entire world as a
+    Load a table of ocean ridge points for the entire world as a
     pandas.DataFrame.
 
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="ocean_ridge_points")`` and will be removed in
-       v0.9.0.
-
     This is the ``@ridge.txt`` dataset used in the GMT tutorials.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:151]: Please see the comments at https://github.com/GenericMappingTools/pygmt/pull/2301#discussion_r1062048779.
```suggestion
```

**Review by seisman — APPROVED** [review]
Looks good to me.

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/samples.py [hunk 2]
```diff
@@ -125,39 +125,23 @@ def _load_japan_quakes():
...
-        warnings.warn(
-            "This function has been deprecated since v0.6.0 and will be removed "
-            "in v0.9.0. Please use load_sample_data(name='ocean_ridge_points') "
-            "instead.",
-            category=FutureWarning,
-            stacklevel=2,
-        )
-
     fname = which("@ridge.txt", download="c")
     data = pd.read_csv(
```
Comment (seisman) [inline pygmt/datasets/samples.py:139]: The two lines:
```
data = read_csv(xxx)
return data
```
can be shortened to `return read_csv(xxx)`.

```suggestion
    return pd.read_csv(
```

File: pygmt/datasets/samples.py [hunk 3]
```diff
@@ -125,39 +125,23 @@ def _load_japan_quakes():
...
-        fname, sep=r"\s+", names=["longitude", "latitude"], skiprows=1, comment=">"
+        fname,
+        delim_whitespace=True,
+        names=["longitude", "latitude"],
+        skiprows=1,
+        comment=">",
     )
     return data
```
Comment (seisman) [inline pygmt/datasets/samples.py:146]:
```suggestion
```

**Review by seisman — APPROVED** [review]

**Pull request merged**